pkg/filter/accesslog/access_log_filter.go

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package accesslog
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+import (
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/constant"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/common/extension"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/config"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/context/http"
+	"github.com/dubbogo/dubbo-go-proxy/pkg/logger"
+	_ "github.com/dubbogo/dubbo-go-proxy/pkg/model"
+)
+
+const (
+	// nolint
+	FileDateFormat = "2006-01-02"
+	// nolint
+	MessageDateLayout = "2006-01-02 15:04:05"
+	// nolint
+	LogFileMode = 0600
+
+	// console
+	Console = "console"
+)
+
+func init() {
+	extension.SetFilterFunc(constant.AccessLogFilter, Filter())
+}
+
+func Filter() context.FilterFunc {
+	return func(c context.Context) {
+		alc := config.GetBootstrap().StaticResources.AccessLogConfig
+		if alc.Enable {
+			start := time.Now()
+			c.Next()
+			latency := time.Now().Sub(start)
+			// build access_log message
+			accessLogMsg := buildAccessLogMsg(c, latency)
```

Comment:
Creating `accessLogMsg`  once

Reply:
i don't quite understand what that means